src/shims/sync.rs

```diff
@@ -323,7 +348,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             mutexattr_get_kind(this, attr_op)?.not_undef()?
         };
 
-        mutex_set_locked_count(this, mutex_op, Scalar::from_u32(0))?;
+        let _ = mutex_get_or_create_id(this, mutex_op)?;
```

Comment:
So we unconditionally create a MutexId on mutex creation? Why do we even need a sentinel value then, we could just move the "create mutex ID" code here into `pthread_mutex_init`, right?

Reply:
Oh wait, we probably cannot use anything but 0 as the default value because of the static initializers...

But then why not still write a 0 here, to make sure this and static initializers use the same code path?